projects/OG-Analytics/src/main/java/com/opengamma/analytics/financial/forex/method/EmptyFXMatrix.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.analytics.financial.forex.method;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.opengamma.util.money.Currency;
+
+/**
+ * An immutable and empty FXMatrix.
+ */
+public class EmptyFXMatrix extends FXMatrix {
+  private static final long serialVersionUID = 1L;
+  private static final double[][] RATES_MATRIX = new double[0][0];
+  private static final Map<Currency, Integer> CURRENCIES_MAP = Collections.emptyMap();
+  
+  /**
+   * Singleton to avoid creating this unnecessarily.
+   */
+  public static final EmptyFXMatrix INSTANCE = new EmptyFXMatrix();
+  
+  public EmptyFXMatrix() {
```

Comment:
Perhaps make this private to enforce use of the singleton instance, then also no need to override hashCode and equals.